### DISCUSSION TITLE
Grep and show only a list of matching commits

### DISCUSSION THREAD

**Discussion by kaihendry** [jesseduffield/lazygit#2310] [Ideas]
How do I view all the commits relating to a particular string (like a bug ID or topic)?

For example in tig: `git rev-list --grep=unicode HEAD | tig show --stdin`

I know there is search, but I don't want the other commits to distract.

Does lazygit have some stdin mode?

**Comment by mark2185** [comment]
I don't think it's currently possible to filter based on `commit` message or `author`, only by `filename` (press `ctrl-s` in the `Commits` panel).

Alternatively, you could run just that as a custom command, press `:` and it will bring a pop-up where you can then type `git rev-list --grep=unicode HEAD | tig show --stdin` if you so desire.

**Reply by jesseduffield** [reply]
I think it wouldn't be too hard to implement a stdin mode, so this is worth promoting to an issue
